### PR TITLE
Feature/authentication

### PR DESCRIPTION
**Pull request by Mnw2212**
Login, session check and logout works.
Angular code along with angularfire is used

### CODE REVIEW

**Comment by Mnw2212** [comment]
@skeep  could you review this ?

File: src/app/index.module.js [hunk 1]
```diff
@@ -4,12 +4,13 @@ import { config } from './index.config';
...
 import { runBlock } from './index.run';
 import { MainController } from './main/main.controller';
+import { LoginController } from './login/login.controller';
 import { GithubContributorService } from '../app/components/githubContributor/githubContributor.service';
 import { WebDevTecService } from '../app/components/webDevTec/webDevTec.service';
 import { NavbarDirective } from '../app/components/navbar/navbar.directive';
 import { MalarkeyDirective } from '../app/components/malarkey/malarkey.directive';
 
-angular.module('vyapi', ['ui.router', 'toastr'])
+angular.module('vyapi', ['ui.router', 'toastr',"firebase"])
```
Comment (skeep) [inline src/app/index.module.js]: use `'` instead of `"`

File: src/app/login/login.controller.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
+export class LoginController {
+  constructor ($scope,$firebaseObject,$firebaseAuth) {
```
Comment (skeep) [inline src/app/login/login.controller.js]: do not use `$scope`. use `this` keyword instead

File: src/app/login/login.controller.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    'ngInject';
+    this.test="Good Morning";
+    this.ref = new Firebase("https://vyapi.firebaseio.com");
+
+    this.firebaseAuthLogin = function(){
```
Comment (skeep) [inline src/app/login/login.controller.js]: use [arrow function](https://babeljs.io/docs/learn-es2015/#arrows-and-lexical-this) here

File: src/app/login/login.html [hunk 4]
```diff
@@ -0,0 +1,9 @@
+<div ng-controller="LoginController as login" class="container">
```
Comment (skeep) [inline src/app/login/login.html]: use `index.route.js` to define controller

File: src/login.html [hunk 5]
```diff
@@ -0,0 +1,65 @@
+<html>
```
Comment (skeep) [inline src/login.html]: delete this fiel

**Pull request merged**